### PR TITLE
Update to Go 1.23

### PR DESCRIPTION
**Pull request by BBBmau**
We'll want to update the wiki also since we recently had the go rewrite https://github.com/hashicorp/terraform-provider-google/wiki/FAQ#-how-do-i-update-the-go-version-used-by-the-terraform-providers-google-and-google-beta

The following will be merged also:
https://github.com/GoogleCloudPlatform/magic-modules/pull/12175
https://github.com/GoogleCloudPlatform/terraform-google-conversion/pull/2995

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:none

```

### CODE REVIEW

File: tpgtools/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/GoogleCloudPlatform/magic-modules/tpgtools
 
-go 1.21
+go 1.22
```
Comment (c2thorn) [inline tpgtools/go.mod:3]: is 1.22 intended instead of 1.23?

Comment (BBBmau) [reply tpgtools/go.mod]: this was a mistake on my part. Must've not saved, thanks for catching this

**Comment by BBBmau** [comment]
unit-test-magician error is unrelated to 1.23 upgrade

```hcl
=== NAME  TestFormatReminderComment/waitingForReview_three_days_with_current_reviewer
    scheduled_pr_reminders_test.go:1084: 
        	Error Trace:	/home/runner/work/magic-modules/magic-modules/.ci/magician/cmd/scheduled_pr_reminders_test.go:1084
        	Error:      	"@SarahFrench This PR has been waiting for review for 3 weekdays. Please take a look! Use the label `disable-review-reminders` to disable these notifications." should not contain "@SarahFrench"
        	Test:       	TestFormatReminderComment/waitingForReview_three_days_with_current_reviewer
=== CONT  TestFormatReminderComment/waitingForReview_two_weeks
=== CONT  TestFormatReminderComment/waitingForMerge_one_week
=== CONT  TestFormatReminderComment/waitingForMerge_two_weeks
=== CONT  TestFormatReminderComment/waitingForContributor_six_weeks
=== CONT  TestFormatReminderComment/waitingForContributor_seven_weeks
=== CONT  TestFormatReminderComment/waitingForContributor_four_weeks
=== CONT  TestFormatReminderComment/waitingForContributor_28_days
=== CONT  TestFormatReminderComment/waitingForReview_one_week
=== CONT  TestFormatReminderComment/waitingForContributor_two_weeks
--- FAIL: TestFormatReminderComment (0.00s)
    --- PASS: TestFormatReminderComment/waitingForReview_three_days (0.00s)
    --- FAIL: TestFormatReminderComment/waitingForReview_three_days_with_current_reviewer (0.00s)
    --- PASS: TestFormatReminderComment/waitingForMerge_many_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForReview_two_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForMerge_one_week (0.00s)
    --- PASS: TestFormatReminderComment/waitingForMerge_two_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForContributor_six_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForContributor_seven_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForContributor_four_weeks (0.00s)
    --- PASS: TestFormatReminderComment/waitingForContributor_28_days (0.00s)
    --- PASS: TestFormatReminderComment/waitingForReview_one_week (0.00s)
    --- PASS: TestFormatReminderComment/waitingForContributor_two_weeks (0.00s)
```

File: tpgtools/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/GoogleCloudPlatform/magic-modules/tpgtools
 
-go 1.21
+go 1.23
```
Comment (c2thorn) [inline tpgtools/go.mod:3]: does tpgtools/go.mod also need to be updated?

Comment (BBBmau) [reply tpgtools/go.mod:3]: running `go mod tidy` didn't produce any dependency changes on go.sum so no update is needed.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**